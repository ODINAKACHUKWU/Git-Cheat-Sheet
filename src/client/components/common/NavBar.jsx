import React from 'react';
import PropTypes from 'prop-types';
import ReactTooltip from 'react-tooltip';

import logo from '../../assets/images/cheat_logo.png';

const NavbarPage = ({ logout, username }) => (
  <div className="d-flex flex-row justify-content-between align-items-center mt-4">
    <img src={logo} alt="app logo" width="40px" height="40px" />
    <h4 style={{ color: '#555' }}>Git Cheat Sheet</h4>
    <button
      type="button"
      data-tip="Log Out"
      onClick={() => logout()}
      style={{
        borderRadius: '50%',
        width: '40px',
        height: '40px',
        textAlign: 'center',
        lineHeight: '40px',
        color: 'white',
        fontSize: '22px',
        backgroundColor: '#bbb',
        cursor: 'pointer',
        outline: 'none'
      }}
    >
      {username[0].toUpperCase()}
    </button>
    <ReactTooltip place="bottom" effect="solid" />
  </div>
);

NavbarPage.propTypes = {
  username: PropTypes.string.isRequired,
  logout: PropTypes.func.isRequired
};

export default NavbarPage;
