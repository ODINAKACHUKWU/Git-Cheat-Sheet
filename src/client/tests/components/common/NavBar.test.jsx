import NavBar from '../../../components/common/NavBar';

describe('<NavBar /> component', () => {
  const props = {
    logout: jest.fn(),
    username: 'testname'
  };
  const shallowWrapper = shallow(<NavBar {...props} />);

  it('renders the header component without crashing', () => {
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('should have two children', () => {
    expect(shallowWrapper.find('div').props().children).toHaveLength(4);
  });
  it('should have a button with text `T` as initials of currently signed in user', () => {
    expect(shallowWrapper.find('button').text()).toEqual(
      props.username[0].toUpperCase()
    );
  });
  it('should call `logout` function when button is clicked', () => {
    shallowWrapper.find('button').simulate('click');
    expect(props.logout).toHaveBeenCalled();
  });
  it('should have a header text of `Git Cheat Sheet`', () => {
    expect(shallowWrapper.find('h4').text()).toEqual('Git Cheat Sheet');
  });
});
