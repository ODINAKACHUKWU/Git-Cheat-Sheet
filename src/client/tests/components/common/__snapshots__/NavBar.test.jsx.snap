// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> component renders the header component without crashing 1`] = `
<div
  className="d-flex flex-row justify-content-between align-items-center mt-4"
>
  <img
    alt="app logo"
    height="40px"
    src={
      Object {
        "process": [Function],
      }
    }
    width="40px"
  />
  <h4
    style={
      Object {
        "color": "#555",
      }
    }
  >
    Git Cheat Sheet
  </h4>
  <button
    data-tip="Log Out"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#bbb",
        "borderRadius": "50%",
        "color": "white",
        "cursor": "pointer",
        "fontSize": "22px",
        "height": "40px",
        "lineHeight": "40px",
        "outline": "none",
        "textAlign": "center",
        "width": "40px",
      }
    }
    type="button"
  >
    T
  </button>
  <ReactTooltip
    effect="solid"
    insecure={true}
    place="bottom"
    resizeHide={true}
    wrapper="div"
  />
</div>
`;
